web/components/tipper.tsx

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import {
+  ChevronDoubleRightIcon,
+  ChevronLeftIcon,
+  ChevronRightIcon,
+} from '@heroicons/react/solid'
+import clsx from 'clsx'
+import { Comment } from 'common/comment'
+import { User } from 'common/user'
+import { formatMoney } from 'common/util/format'
+import { debounce, sumBy } from 'lodash'
+import { useEffect, useMemo, useRef, useState } from 'react'
+import { CommentTips } from 'web/hooks/use-comment-tips'
+import { useUser } from 'web/hooks/use-user'
+import { transact } from 'web/lib/firebase/fn-call'
+import { Row } from './layout/row'
+import Tooltip from './tooltip'
+
+// xth triangle number * 5  =  5 + 10 + 15 + ... + (x * 5)
+const quad = (x: number) => (5 / 2) * x * (x + 1)
+
+// inverse (see https://math.stackexchange.com/questions/2041988/how-to-get-inverse-of-formula-for-sum-of-integers-from-1-to-nsee )
+const invQuad = (y: number) => Math.sqrt((2 / 5) * y + 1 / 4) - 1 / 2
+
+function Tipper(prop: { comment: Comment; tips: CommentTips }) {
+  const { comment, tips } = prop
+
+  const me = useUser()
+  const myId = me?.id ?? ''
+  const savedTip = tips[myId] as number | undefined
+
+  // optimistically increase the tip count, but debounce the update
+  const [localTip, setLocalTip] = useState(savedTip ?? 0)
+  const initialized = useRef(false)
+  useEffect(() => {
+    if (savedTip && !initialized.current) {
+      setLocalTip(savedTip)
+      initialized.current = true
+    }
+  }, [savedTip])
+
+  const score = useMemo(() => {
+    const tipVals = Object.values({ ...tips, [myId]: localTip })
+    return sumBy(tipVals, invQuad)
```

Comment:
I'm skeptical of showing users a square-root scaled sum. I think they will say it doesn't make sense, especially if they add some M$, and it goes up by less than they paid. We can try it on users, I guess. I give 80% odds we'll switch it to a plain sum though.

Reply:
it makes more sense in the context of sorting by comment karma (which isn't in this PR)

Reply:
I think the actual way people will be confused in practice is thinking that they can only upvote once